### DISCUSSION TITLE
Automatically support vN, vN.M references

### DISCUSSION THREAD

**Discussion by myitcv** [community/community#9847] [Actions]
Currently it appears that in order to support, for example, a reference to `me/myaction@v1`, I have to create a tag for `v1` in my repository. 

This requirement creates a few problems:

* there is nothing that requires `v1` to refer to an actual `v1.x.y` tag. i.e. it is possible for `v1` to refer to something other than a valid `v1.x.y` tag;
* encouraging people to move tags is problematic because it normalises the practice. i.e. if I tag `v1.2.3` but spot a problem, add a new commit, why not just re-tag `v1.2.3`? This has led to many advising that commit shas be used as the version; 
* there is churn work required to move the `v1`, or more generally, `vN` or `vN.M` tags;
* I would posit that the vast majority of `vN` tags refer to the latest `vN.x.y` version (an order defined by [the semver standard](https://semver.org/)), and for those that don't a significant percentage _should_ do but don't by virtue of a `vN` tag not having been updated.

My proposal is therefore that action version resolution logic should be updated as follows:

* if a `vN` or `vN.M` tag exists, use that tag;
* else `vN` should refer to the latest non-pre-release version `vN.x.y`;
* or `vN.M` should refer to the latest non-pre-release version `vN.M.x`.

This proposal is backwards compatible with existing `vN` and `vN.M` tags.

**Comment by umarcor** [comment]
Refs:

- https://github.com/actions/toolkit/pull/631
- https://github.com/actions/toolkit/issues/214

**Comment by ethomson** [comment]
Thanks for the feedback.  I agree that "sliding tags" is not particularly ergonomic.  The team has a few things on their plate at the moment, but this is something that we'll take a look at in the not-too-distant future.

**Comment by shakefu** [comment]
I think while tag parsing is certainly a feasible solution it also involves pulling and parsing the entire list of remote refs, which isn't always desirable.

I would personally love to see first class support for Semantic Releases in GitHub itself, rather than kludging something onto `git`. A new Repository Setting of "Support Semantic Releases" and an API call allowing standard SemVer identifiers to find and download release assets and source would completely support this use case and others. Additionally this could enable GHA to run against release assets in a consistent manner which could be limited to only the necessary runtime, e.g. `dist/index.js` and `action.yml`.

```http
GET /repos/{owner}/{repo}/releases/semver/{semver}
```

I realize that's a lot more effort than just changing the Action runner to be able to do basic tag parsing, though.
